tests/end-to-end/director/labeling_api_test.go

```diff
@@ -0,0 +1,845 @@
+	require.NoError(t, err)
+	createReq = gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+  				result: createApplication(in: %s) {
+    					%s
+					}
+				}`, appInputGQL, tc.gqlFieldsProvider.ForApplication()))
+	secondApp := graphql.Application{}
+
+	//WHEN
+	err = tc.RunQuery(ctx, createReq, &secondApp)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstApp.ID)
+	defer deleteApplication(t, firstApp.ID)
+
+	//Set label "foo" on both applications
+	// GIVEN
+	labelKeyFoo := "foo"
+	labelValue := "val"
+
+	setLabelRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: setApplicationLabel(applicationID: %s, key: %s, value: %v) {
+						%s
+					}
+				}`,
+			firstApp.ID, labelKeyFoo, labelValue, tc.gqlFieldsProvider.ForApplicationLabel()))
+	firstAppLabel := graphql.Label{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, setLabelRequest, &firstAppLabel)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstAppLabel.Key)
+	require.NotEmpty(t, firstAppLabel.Value)
+
+	// GIVEN
+	setLabelRequest = gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: setApplicationLabel(applicationID: %s, key: %s, value: %v) {
+						%s
+					}
+				}`,
+			secondApp.ID, labelKeyFoo, labelValue, tc.gqlFieldsProvider.ForApplicationLabel()))
+	secondAppLabel := graphql.Label{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, setLabelRequest, &secondAppLabel)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstAppLabel.Key)
+	require.NotEmpty(t, firstAppLabel.Value)
+
+	//Set label "bar" on first application
+	// GIVEN
+	labelKeyBar := "bar"
+	labelValue = "barval"
+
+	setLabelRequest = gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: setApplicationLabel(applicationID: %s, key: %s, value: %v) {
+						%s
+					}
+				}`,
+			firstApp.ID, labelKeyBar, labelValue, tc.gqlFieldsProvider.ForApplicationLabel()))
+	firstAppBarLabel := graphql.Label{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, setLabelRequest, &firstAppBarLabel)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstAppLabel.Key)
+	require.NotEmpty(t, firstAppLabel.Value)
+
+	// Query for application with LabelFilter "foo"
+	// GIVEN
+	lf := graphql.LabelFilter{
+		Key:   labelKeyFoo,
+		Query: nil,
+	}
+
+	lfGQL, err := tc.graphqlizer.LabelFilterToGQL(lf)
+	require.NoError(t, err)
+
+	getApps := gcli.NewRequest(
+		fmt.Sprintf(`query {
+				result: applications(filter: %s, first: %d, after: %s) {
+						%s
+					}
+				}`,
+			lfGQL, 5, "", tc.gqlFieldsProvider.Page(tc.gqlFieldsProvider.ForApplication())))
+
+	appPage := graphql.ApplicationPage{}
+	// WHEN
+	err = tc.RunQuery(ctx, getApps, appPage)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, appPage)
+	assert.Equal(t, appPage.TotalCount, 2)
+	assert.Contains(t, appPage.Data[0].Labels, labelKeyFoo)
+	assert.Equal(t, appPage.Data[0].Labels[labelKeyFoo], labelValue)
+	assert.Contains(t, appPage.Data[1].Labels, labelKeyFoo)
+	assert.Equal(t, appPage.Data[1].Labels[labelKeyFoo], labelValue)
```

Comment:
Yup, as @tgorgol said, don't do this - if you see accessing different values from array like `[0], [1]`, it means that it has to be converted to loops